orchagent/copporch.cpp

```diff
@@ -125,7 +125,7 @@ void CoppOrch::initDefaultHostIntfTable()
     if (status != SAI_STATUS_SUCCESS)
     {
         SWSS_LOG_ERROR("Failed to create default host interface table, rv:%d", status);
-        throw "CoppOrch initialization failure";
+        handleSaiCreateStatus(SAI_API_HOSTIF, status);
```

Comment:
>handleSaiCreateStatus [](start = 8, length = 21)

Do you want to keep old behavior even with your new handle function?

Reply:
I think in the old behavior (async mode), this throw should not even be called since syncd should already trigger an swss restart before it reaches this throw. To mimic the old behavior close enough, I think a method could be running this handling function before this throw and keep this throw regardless of the return value of the handling function. I updated the code as such.

Reply:
You also need to check return value and conditional `throw`.

---
In reply to: [595610233](https://github.com/Azure/sonic-swss/pull/1665#discussion_r595610233) [](ancestors = 595610233)

Reply:
Updated the logic to check return value before throwing.